File: FestivalJava/src/main/java/Repository/Interfaces/IUserRepository.java
package Repository.Interfaces;

import Domain.User;

/**
 * Created by Sebi on 26-Mar-17.
 */
public interface IUserRepository extends IRepository<User, String> {
    /*
    Returns a user entity if login is possible
    else null
     */
    User getUserForLogin(String username, String password);
}

File: FestivalC#/Festival/Repository/Repository/Interface/IUserRepository.cs
using Festival.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Repository.Interface
{
    public interface IUserRepository : IRepository<User, String>
    {
        /*
         * Used for login
         */
        User getUserFor(String username, String password);
    }
}
